src/snowflake/snowpark/mock/_snowflake_data_type.py

```diff
@@ -330,6 +339,7 @@ def __add__(self, other):
         result = super().__add__(other)
         if self.sf_type:
             result.sf_type = calculate_type(self.sf_type, other.sf_type, op="+")
+        result = reset_nan_to_none(self, other, result)
```

Comment:
But what if we want to have `nan` in the result?

Reply:
I'm good with the name change